diff --git a/service/src/com/android/car/hal/fakevhal/FakeVhalConfigParser.java b/service/src/com/android/car/hal/fakevhal/FakeVhalConfigParser.java
old mode 100644
new mode 100755
index 1acc965..8d882df
--- a/service/src/com/android/car/hal/fakevhal/FakeVhalConfigParser.java
+++ b/service/src/com/android/car/hal/fakevhal/FakeVhalConfigParser.java
@@ -208,7 +208,19 @@ public final class FakeVhalConfigParser {
             Map.entry("PLACEHOLDER_PROPERTY_BOOLEAN",
                     TestVendorProperty.PLACEHOLDER_PROPERTY_BOOLEAN),
             Map.entry("PLACEHOLDER_PROPERTY_STRING",
-                    TestVendorProperty.PLACEHOLDER_PROPERTY_STRING)
+                    TestVendorProperty.PLACEHOLDER_PROPERTY_STRING),
+           Map.entry("VENDOR_EXTENSION_Light_CONTROL_PROPERTY",
+                    TestVendorProperty.VENDOR_EXTENSION_Light_CONTROL_PROPERTY),
+            Map.entry("VENDOR_EXTENSION_BATERRY_PROPERTY",
+                    TestVendorProperty.VENDOR_EXTENSION_BATERRY_PROPERTY), 
+            Map.entry("VENDOR_EXTENSION_FLeft_Door_PROPERTY",
+                    TestVendorProperty.VENDOR_EXTENSION_FLeft_Door_PROPERTY), 
+            Map.entry("VENDOR_EXTENSION_RLeft_Door_PROPERTY",
+                    TestVendorProperty.VENDOR_EXTENSION_RLeft_Door_PROPERTY), 
+            Map.entry("VENDOR_EXTENSION_FRight_Door_PROPERTY",
+                    TestVendorProperty.VENDOR_EXTENSION_FRight_Door_PROPERTY), 
+            Map.entry("VENDOR_EXTENSION_RRight_Door_PROPERTY",
+                    TestVendorProperty.VENDOR_EXTENSION_RRight_Door_PROPERTY) 
     );
 
     /**
